src/urllib3/poolmanager.py

```diff
@@ -360,7 +407,7 @@ def urlopen(self, method, url, redirect=True, **kw):
             return response
 
         # Support relative URLs for redirecting.
-        redirect_location = urljoin(url, redirect_location)
+        redirect_location = str(urljoin(url, redirect_location))
```

Comment:
Why does this need a `str()` call? Shouldn't `urljoin(str, str)` be `str`?

Reply:
> Shouldn't urljoin(str, str) be str?

Yes

> Why does this need a str() call? 

Because the type of `redirect_location` is `Union[bool, str]`  https://github.com/urllib3/urllib3/blob/2ef05adfcf001c85f1f95ae3fa078d16190851b0/src/urllib3/response.py#L202

I will change this line to 
```
redirect_location = urljoin(url, str(redirect_location))
```